actionmailer/lib/action_mailer/railtie.rb

```diff
@@ -46,9 +45,13 @@ class Railtie < Rails::Railtie # :nodoc:
           self.delivery_job = delivery_job.constantize
         end
 
+        if options.smtp_settings
+          self.smtp_settings = options.smtp_settings
+        end
+
         if smtp_timeout = options.delete(:smtp_timeout)
```

Comment:
Isn't this going to raise an exception when `options.smtp_settings` is not present? Should not this be moved to inside that conditional?

Reply:
It shouldn't fail when `options.smtp_settings` is not present because ActionMailer::Base does have a [default value](https://github.com/rails/rails/blob/main/actionmailer/lib/action_mailer/delivery_methods.rb#L20-L27). 

So if `options.smtp_settings` is nil this code basically adds smtp_timeout for the default smtp_settings from ActionMailer::Base. I've added a test for that.

Although It could raise an error if someone does `ActionMailer::Base.smtp_settings = nil` before Rails loads. So, I addressed that case and added another test to ensure it does not raise.

Reply:
But we are deleting from `options` not `self.smtp_settings`, so the default value doesn't apply.

Reply:
Forget about it. I'm confused.